processor/otel/consumer.go

```diff
@@ -58,3 +74,437 @@ func (c *Consumer) ConsumeTraceData(ctx context.Context, td consumerdata.TraceDa
+			case "component":
+				component = truncate(v.StringValue.Value)
+				fallthrough
+			default:
+				utility.DeepUpdate(labels, k, truncate(v.StringValue.Value))
+			}
+		}
+	}
+
+	if event.Type == "" {
+		if isHTTP {
+			event.Type = "request"
+		} else if component != "" {
+			event.Type = component
+		} else {
+			event.Type = "custom"
+		}
+	}
+
+	if isHTTP {
+		if code := int(span.GetStatus().GetCode()); code != 0 {
+			result = statusCodeResult(code)
+			if http.Response == nil {
+				http.Response = &model.Resp{StatusCode: &code}
+			}
+		}
+		event.Http = &http
+	}
+
+	if result == "" {
+		if hasFailed {
+			result = "Error"
+		} else {
+			result = "Success"
+		}
+	}
+	event.Result = &result
+
+	if len(labels) == 0 {
+		return
+	}
+	l := model.Labels(labels)
+	event.Labels = &l
+}
+
+func parseSpan(span *tracepb.Span, event *model_span.Event) {
+	labels := make(common.MapStr)
+
+	var http model_span.HTTP
+	var db model_span.DB
+	var destination model_span.Destination
+	var isDBSpan, isHTTPSpan bool
+	var component string
+	for kDots, v := range span.Attributes.GetAttributeMap() {
+		k := replaceDots(kDots)
+		switch v := v.Value.(type) {
+		case *tracepb.AttributeValue_BoolValue:
+			utility.DeepUpdate(labels, k, v.BoolValue)
+		case *tracepb.AttributeValue_DoubleValue:
+			utility.DeepUpdate(labels, k, v.DoubleValue)
+		case *tracepb.AttributeValue_IntValue:
+			switch kDots {
+			case "http.status_code":
+				//v.IntValue is int64, considering the semantics of the field converting to an int should be fine
+				code := int(v.IntValue)
+				http.StatusCode = &code
+				isHTTPSpan = true
+			case "peer.port":
+				port := int(v.IntValue)
+				destination.Port = &port
+			default:
+				utility.DeepUpdate(labels, k, v.IntValue)
+			}
+		case *tracepb.AttributeValue_StringValue:
+			switch kDots {
+			case "span.kind": // filter out
+			case "http.url":
+				url := truncate(v.StringValue.Value)
+				http.URL = &url
+				isHTTPSpan = true
+			case "http.method":
+				method := truncate(v.StringValue.Value)
+				http.Method = &method
+				isHTTPSpan = true
+			case "sql.query":
+				// NOTE(axw) in theory we could override
+				// the span name here with our own method
+				// based on the SQL query.
+				db.Statement = &v.StringValue.Value
+				if db.Type == nil {
+					dbType := "sql"
+					db.Type = &dbType
+				}
+				isDBSpan = true
+			case "db.statement":
+				db.Statement = &v.StringValue.Value
+				isDBSpan = true
+			case "db.instance":
+				db.Instance = &v.StringValue.Value
+				isDBSpan = true
+			case "db.type":
+				db.Type = &v.StringValue.Value
+				isDBSpan = true
+			case "db.user":
+				db.UserName = &v.StringValue.Value
+				isDBSpan = true
+			case "peer.address":
+				val := truncate(v.StringValue.Value)
+				destination.Address = &val
+			case "peer.service":
+				val := truncate(v.StringValue.Value)
+				event.DestinationService = &model_span.DestinationService{Name: &val}
```

Comment:
We'll have to make sure this doesn't break service maps, because at the moment the agents are required to send all or none of `destination.service.{name,type,resource}`.

Reply:
Good point, I will remove the logic and only add to labels, as it doesn't make sense to store the information here when not usable for service maps.